include/nbl/asset/utils/IMeshManipulator.h

```diff
@@ -451,11 +457,32 @@ class IMeshManipulator : public virtual core::IReferenceCounted
 		}
 
 		//! Recalculates the cached bounding box of the meshbuffer
-		static inline void recalculateBoundingBox(ICPUMeshBuffer* meshbuffer)
+		static inline void recalculateBoundingBox(ICPUMeshBuffer* meshbuffer, bool isOBBDisabled)
 		{
-			meshbuffer->setBoundingBox(calculateBoundingBox(meshbuffer,meshbuffer->getJointAABBs()));
+      if(isOBBDisabled)
+      { meshbuffer->setBoundingBox(calculateBoundingBox(meshbuffer,meshbuffer->getJointAABBs())); }
+      else
+      { meshbuffer->setBoundingBox(calculateOrientedBBox(meshbuffer)); }
 		}
 
+    static inline core::OBB calculateOrientedBBox(const ICPUMeshBuffer* meshBuffer)
+    {
+      auto vtxCount = meshBuffer->getIndexCount();
+
+      std::vector<core::vectorSIMDf> vtxList;
+      vtxList.reserve(vtxCount);
+
+      for(auto i = 0u; i < vtxCount; i++)
+      { vtxList.push_back(meshBuffer->getPosition(i)); }
```

Comment:
this will trigger an "out of bounds" assert for most indexed meshes, go and see how the `calculateBoundingBox` function iterates through a meshbuffer's vertices

Reply:
no it doesn't